### DISCUSSION TITLE
Runner scaleset not picking up the jobs even though the runner is online

### DISCUSSION THREAD

**Discussion by ChaitanyaAtchuta5** [actions/actions-runner-controller#2932] [Questions]
I am testing out new ARC. So I have created a simple Workflow that has 3 jobs in parallel(They just sleep for 30sec and print Hello world).  The issue I am facing is some times one of the job is not even though the new runner is available on runner scale set.
Note: I have seen this issue when using traditional way of self-hosted runners

Version: 0.6.1 (I saw this behavior in 0.6.0 as well)

WF Job status
<img width="1670" alt="image" src="https://github.com/actions/actions-runner-controller/assets/68017180/ecf80589-f56c-4d59-be3b-9ded0b7baf3e">

<img width="345" alt="image" src="https://github.com/actions/actions-runner-controller/assets/68017180/80deae19-3c7b-461d-a891-84e84a4663f2">

Logs of pod that was not picking up the jobs
`➜  ~ kubectl logs -n github-runners arc-runner-set-4spwn-runner-stmbr -c runner
2023-09-26 18:28:51Z: Runner connect error: Connection refused (pipelinesghubeus4.actions.githubusercontent.com:443). Retrying until reconnected.

√ Connected to GitHub

2023-09-26 18:29:24Z: Runner reconnected.
Current runner version: '2.308.0'
2023-09-26 18:29:24Z: Listening for Jobs
2023-09-26 18:29:28Z: Running job: HelloWorld3
An error occurred: Access denied. System:ServiceIdentity;DDDDDDDD-DDDD-DDDD-DDDD-DDDDDDDDDDDD needs View permissions to perform the action.
Runner listener exit with retryable error, re-launch runner in 5 seconds.
Restarting runner...

√ Connected to GitHub

2023-09-26 18:30:28Z: Runner connect error: One or more errors occurred. (Registration abf9c583-8ce3-4bcb-ae0a-a3823a9ac8bc was not found.). Retrying until reconnected.

√ Connected to GitHub

√ Connected to GitHub

√ Connected to GitHub

√ Connected to GitHub
`
Also, we are using istio proxy and an extra container "istio-proxy" gets attached to the pod.

**Comment by thisisthetechie** [comment]
> An error occurred: Access denied. System:ServiceIdentity;DDDDDDDD-DDDD-DDDD-DDDD-DDDDDDDDDDDD needs View permissions to perform the action.

Looks like your system account is missing some permissions...
